components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/ActionExecutorConfig.java

```diff
@@ -73,6 +80,35 @@ public boolean isExecutionForActionTypeEnabled(ActionType actionType) {
         }
     }
 
+    public int getHttpReadTimeout() {
+
+        return parseTimeoutConfig(HTTP_READ_TIMEOUT_PROPERTY, DEFAULT_HTTP_READ_TIMEOUT);
+    }
+
+    public int getHttpConnectionRequestTimeout() {
+
+        return parseTimeoutConfig(HTTP_CONNECTION_REQUEST_TIMEOUT_PROPERTY, DEFAULT_HTTP_CONNECTION_REQUEST_TIMEOUT);
+    }
+
+    public int getHttpConnectionTimeout() {
+
+        return parseTimeoutConfig(HTTP_CONNECTION_TIMEOUT_PROPERTY, DEFAULT_HTTP_CONNECTION_TIMEOUT);
+    }
+
+    private int parseTimeoutConfig(String timeoutTypeName, int defaultTimeout) {
+
+        int timeoutPropertyValue = defaultTimeout;
+        String timeoutValue = (String) IdentityConfigParser.getInstance().getConfiguration().get(timeoutTypeName);
+        if (StringUtils.isNotBlank(timeoutValue)) {
+            try {
+                timeoutPropertyValue = Integer.parseInt(timeoutValue);
+            } catch (Exception e) {
```

Comment:
Let's not catch for exception. This is supposed to thrown a parsing error right ?
Let's capture that specific runtime exception here

Reply:
Updated in https://github.com/wso2/carbon-identity-framework/pull/5935/commits/e89ea4c2346f44139605a36f742988f5a0e4e84b